migration/migrator.go

```diff
@@ -215,28 +215,29 @@ func (m *StateMigrator) migrateStorage(
 		return common.Hash{}, err
 	}
 
-	var mu sync.Mutex
+	nodeIt, err := zkt.NodeIterator(nil)
+	if err != nil {
+		return common.Hash{}, fmt.Errorf("failed to open node iterator (root: %s): %w", zkt.Hash(), err)
+	}
+	iter := trie.NewIterator(nodeIt)
 	var slots atomic.Uint64
-	err = hashRangeIterator(m.ctx, zkt, NumProcessStorage, func(key, value []byte) error {
-		hk := trie.IteratorKeyToHash(key, true)
+	for iter.Next() {
+		hk := trie.IteratorKeyToHash(iter.Key, true)
 		preimage, err := m.readZkPreimage(*hk)
 		if err != nil {
-			return err
+			return common.Hash{}, err
 		}
 		slot := common.BytesToHash(preimage).Bytes()
-		trimmed := common.TrimLeftZeroes(common.BytesToHash(value).Bytes())
-		mu.Lock()
-		defer mu.Unlock()
+		trimmed := common.TrimLeftZeroes(common.BytesToHash(iter.Value).Bytes())
 		if err := mpt.UpdateStorage(address, slot, trimmed); err != nil {
-			return err
+			return common.Hash{}, err
 		}
 
 		slots.Add(1)
-		log.Trace("Updated storage slot to MPT", "contract", address.Hex(), "index", common.BytesToHash(key).Hex())
-		return nil
-	})
-	if err != nil {
-		return common.Hash{}, err
+		log.Trace("Updated storage slot to MPT", "contract", address.Hex(), "index", common.BytesToHash(iter.Key).Hex())
+	}
+	if iter.Err != nil {
```

Comment:
Question here, if the above for loop ends, won't this `iter.Err` be always not nil? Is some kind of adding a handling for `errIteratorEnd` needed here?

Reply:
if the loop ends as expected, iter.Err will be nil. However, if the loop ends unexpectedly, iter.Err will contain a seekError